apps/dashboard/lib/trpc/trpc.ts

```diff
@@ -17,3 +19,38 @@ export const auth = t.middleware(({ next, ctx }) => {
     },
   });
 });
+
+export const protectedProcedure = t.procedure.use(auth);
```

Comment:
what's the benefit of this way over defining middlewares as needed in each handler?

Reply:
If I understand this correctly. Seperate Procedures makes it easier to understand the difference between ratelimit and authenticated procedures.

Reply:
sorry I meant compared to the previous way where in each handler file we did:
```ts
t.procedude
  .use(auth)
  .input(...)
```

It could be extended with ratelimiting like so:
```ts
t.procedude
  .use(auth)
  .use(ratelimit(opts))
  .input(...)
```

Why do you prefer to do it like you did? Not saying it's wrong, just curious about the reasoning behind it

Reply:
>  Seperate Procedures makes it easier to understand the difference between ratelimit and authenticated procedures.

it's not actually separate though, how are you now going to create a ratelimited but not authenticated route?
I'm not even sure we need this, just trying to think through

Reply:
The main benefit I can see using it this way instead of middleware: It is clearer when used for new people developing on the platform, ratelimitProcedure indicates this route should use ratelimiting. This could be just opinion on my part. But to me it seems simpler to use and not have to worry as much about options passed. and remembering to protectedRoute.use(ratelimit(opts)). We always want to use an authenticated request to ratelimit as we are using the user for process the ratelimit checks, I believe there should never be a scenario where we aren’t using an authenticated route where ratelimit is included.

Reply:
makes sense, thanks